docs/topics/templates.txt

```diff
@@ -415,6 +415,22 @@ adds defaults that differ from Jinja2's for a few options:
 
   It defaults to an empty list.
 
+  Use context processors for non-trivial request-dependent values that you want
+  computed only once per template render (and not at all if no template is
+  rendered).
+
+  If a context processor's value isn't used in a majority of your templates or
+  isn't used more than once per template render, it would be more explicit to
+  put the function that you would use as a context processor in the template
+  context (for example, using ``jinja2.Environment`` as described below) and
+  then to call that function in the template::
+
+    {{ function(request) }}
+
+  Since context processors are run for every template that's rendered, think
+  carefully before putting expensive computations in them, especially if the
+  result isn't used in all or most of your templates.
+
```

Comment:
Looks mostly good but isn't the point about "Not wanting to cache the result in an attribute of the request, for purity." really more like "Avoiding the computation on requests that don't render templates." (assuming the alternative is caching the value on the request in a middleware's `process_request()`). I think that option should be described -- right now it looks like there are only two options: context processors and passing the function to the template.